### ISSUE TITLE
ERC: registry for token symbol?

### ISSUE THREAD

**Issue by bernardpeh** [ethereum/EIPs#740]
People always want to have sexy token symbol when launching ico. There is nothing stopping anyone from using a name that has been used before. Parity has a token registry standard - https://github.com/paritytech/parity/wiki/Token-Registry but its still not widely accepted.

I can think of 2 issues for not having central governance of token symbols.

1. We are relying on people to check https://etherscan.io/tokens when registering a new token and create a new one that has not been used. There is nothing stopping anyone from spamming 100 new erc20 tokens with OMG symbol for example. It is also possible to create dummy erc20 tokens and park all sexy 3 letters and 4 letters symbols, just like how people park domain names during the .com boom.

2. Hackers might create new contracts using existing token symbol and create fake advertisements asking people to buy the token. Victims follow the instructions to add the token address in MEW and see that the token name is indeed correct, not realising the address is incorrect. This hack has been done in ico before.

Solution: Following the footpath of dns and ens, why not create a token named service (tns) managed by ethereum foundation? Then teach people to always use this service as the source of truth.

**Comment by jojeyh** [comment]
Tokens already have a verifiable unique identity based on the contract address, so the need to implement an official token registry seems redundant.  Unofficial ones could be very useful though.

**Comment by bernardpeh** [comment]
@jojeyh agreed but unregulated token symbol means you could have 100 OMG under 100 different contract addresses. How do you proof the authenticity of the token by merely looking at the address? How do you know the googled website is indeed the real website containing the real address of the token?

**Comment by Arachnid** [comment]
Why not use ENS for this? You can use any subdomain you want to establish your own registry; everyone can choose which registry they want to listen to.

**Comment by bernardpeh** [comment]
I think the gist of this is the protection of the brand. ENS is just an alias for an address and should not be confused with the rightful owner of the brand. Put this in real world analogy, if we have no control over the symbol, we are allowing anyone to operate under the name IBM from any parts of the world. As the original IBM becomes more popular, 1000 more IBM springs up from nowhere profiting from the brand and faking their identity.

**Comment by Shrugs** [comment]
@bernardpeh I think nick means you can get a .eth domain and make your own registry, call it `tns.eth`. And then you do manual, human curation (or add some sort of token-curated registry logic so that it can be governed by the crowd) for pointing each ticker symbol to the correct address (`omg.tns.eth`, `nrm.tns.eth`, etc). No need to build a separate naming system; ENS is general enough.

**Comment by bernardpeh** [comment]
hi @Shrugs thanks for the clarification. Who manage tns.eth then? how do we know that tns.eth is official and the standard?

**Comment by Shrugs** [comment]
@bernardpeh you do! The only want to ensure "brand ownership" (instead of, like, contract address ownership) is to either "get humans involved" to solve the squatting issue or let the community collectively decide who owns what brand (TCR). Not sure there's any other option. And there's no way to make a standard; people would have to adopt it. But if you say "here are 10 public figures in the community and they're the 10 people with the keys to this MultiSig that is the only contract that can change the registry" that's probably enough justification for people to consider it the One True Registry™. Or you could think up some sort of TCR to incentivize people to vote on who owns what brand. I think this could be valuable, especially in the case of contention around token symbols (see the recent AIR debacle, with AirSwap switching to AST); the most support a project has, the more of a claim it has to the brand. Lot of issues with this approach as well (51% attacks, need to start a flywheel, etc). 

Basically, the tech has been figured out by ENS; there's no reason to do any foundational development since the generic concept of "ethereum-based human readable string pointer to something else" has been solved. The real development would be the registrar you build to manage your .ens token registry domain.

**Comment by bernardpeh** [comment]
I also realised etherscan had an unofficial reputation rating against each token. perhaps that could be an alternative for now.
